include/up-cpp/transport/UListener.h

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2024 General Motors GTO LLC
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * 
+ * SPDX-FileType: SOURCE
+ * SPDX-FileCopyrightText: 2024 General Motors GTO LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _ULISTENER_H_
+#define _ULISTENER_H_
+
+#include <up-core-api/ustatus.pb.h>
+#include <up-core-api/uri.pb.h>
+#include <up-core-api/uattributes.pb.h>
+#include <up-cpp/transport/datamodel/UMessage.h>
+#include <up-cpp/transport/datamodel/UPayload.h>
+
+namespace uprotocol::utransport {
+
+    class UListener {
+
+        public:
+            /**
+             * Method called to handle/process events.
+             *
+             * @param message Topic the underlying source of the message.
+             *
+             * @return Returns an Ack every time a message is received
+             *         and processed.
+             */
+            virtual uprotocol::v1::UStatus onReceive(
```

Comment:
I think the UListener base class should be replaced with a std::function<void (const UMessage&)> is more C++ idiomatic. Short of this change, there is no good reason for the UListener class to be declared const, or the onReceive method.

Reply:
Given the possibilities to have errors during the send phase or the response phase of an RPC call, wouldn't it make more sense to use exceptions uniformly throughout the C++ API. This would make it simpler to implement constructors and factory functions wrapping constructors.

Reply:
Discussed with Greg in call. I'm dropping the point.